stdlib/public/Concurrency/Clock.cpp

```diff
@@ -41,19 +41,15 @@ void swift_get_time(
 #elif (defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__wasi__))
       clock_gettime(CLOCK_MONOTONIC, &continuous);
 #elif defined(_WIN32)
-      LARGE_INTEGER freq;
-      QueryPerformanceFrequency(&freq);
-      LARGE_INTEGER count;
-      QueryPerformanceCounter(&count);
-      // Divide count (number of ticks) by frequency (number of ticks per
-      // second) to get the counter in seconds. We also need to multiply the
-      // count by 1,000,000,000 to get nanosecond resolution. By multiplying
-      // first, we maintain high precision. The resulting value is the tick
-      // count in nanoseconds. Use 128-bit math to avoid overflowing.
-      auto quadPart = static_cast<unsigned _BitInt(128)>(count.QuadPart);
-      auto ns = (quadPart * 1'000'000'000) / freq.QuadPart;
-      continuous.tv_sec = ns / 1'000'000'000;
-      continuous.tv_nsec = ns % 1'000'000'000;
+      // This needs to match what swift-corelibs-libdispatch does
+
+      // QueryInterruptTimePrecise() outputs a value measured in 100ns
```

Comment:
Given how almost-identical this is to the other clock, would it make sense to factor it out and just have a `ULONGLONG_to_timespec()` function?

Reply:
We're only doing this in a couple of places, so I think I'm relaxed about it for now. If we add another one then yes, probably.